workspaces/adventure-pack/goodies/java/src/digits_int/AP.java

```diff
@@ -0,0 +1,45 @@
+package digits_int;
+
+import iterable_int_stream.IterableIntStream;
+import java.util.stream.IntStream;
+
+public final class AP {
+
+  /**
+   * Get the digits in the given integer. The digits are returned as an
+   * {@link IterableIntStream integer stream} with the least significant digit as the first element
+   * in the stream. For example, for integer 27363, [3,6,3,7,2] is returned.
+   * @param number Integer of interest.
+   * @return The {@link IterableIntStream stream} of digits.
+   * @throws Exception In case of negative integers.
+   */
+  public static IterableIntStream digits(int number) throws Exception {
+    return digits(number, 10);
+  }
+
+  /**
+   * Get the digits in the given integer for the given radix value. The digits are returned as an
+   * {@link IterableIntStream integer stream} with the least significant digit as the first element
+   * in the stream. For example, for integer 27363, [3,6,3,7,2] is returned with radix = 10.
+   * @param number Integer of interest.
+   * @param radix Radix value of interest.
+   * @return The {@link IterableIntStream stream} of digits.
+   * @throws Exception In case of negative integers or radix less than 2.
+   */
+  public static IterableIntStream digits(int number, int radix)
+    throws Exception {
+    if (number < 0) throw new Exception("Please provide a positive integer.");
```

Comment:
We could make these IllegalArgumentException instead.

Reply:
Addressed.